From 82b5be84165093c678a7a742a7f53268935d3628 Mon Sep 17 00:00:00 2001
From: Stephan Hartmann <stha09@googlemail.com>
Date: Thu, 22 Apr 2021 16:32:32 +0000
Subject: [PATCH] GCC: remove alias for GetDownloadsCallback

GCC finds two GetDownloadsCallback definitions, one in
DownloadShelfHandler and one in PageHandler. Remove the alias from
DownloadShelfHandler and only use the one from PageHandler.

Bug: 819294
Change-Id: I5fac82368bd42c70e36944cf7ad463d66fad4a8f
Reviewed-on: https://chromium-review.googlesource.com/c/chromium/src/+/2844829
Commit-Queue: Keren Zhu <kerenzhu@chromium.org>
Reviewed-by: Keren Zhu <kerenzhu@chromium.org>
Cr-Commit-Position: refs/heads/master@{#875199}
---

diff --git a/chrome/browser/ui/webui/download_shelf/download_shelf_handler.h b/chrome/browser/ui/webui/download_shelf/download_shelf_handler.h
index 25e6b6b..723cfef 100644
--- a/chrome/browser/ui/webui/download_shelf/download_shelf_handler.h
+++ b/chrome/browser/ui/webui/download_shelf/download_shelf_handler.h
@@ -11,12 +11,10 @@
 
 class DownloadShelfHandler {
  public:
-  using GetDownloadsCallback = base::OnceCallback<void(
-      std::vector<download_shelf::mojom::DownloadItemPtr>)>;
-
   virtual ~DownloadShelfHandler() = default;
 
-  virtual void GetDownloads(GetDownloadsCallback callback) = 0;
+  virtual void GetDownloads(
+      download_shelf::mojom::PageHandler::GetDownloadsCallback callback) = 0;
 
   virtual void ShowContextMenu(uint32_t download_id,
                                int32_t client_x,
diff --git a/chrome/browser/ui/webui/download_shelf/download_shelf_ui_unittest.cc b/chrome/browser/ui/webui/download_shelf/download_shelf_ui_unittest.cc
index 7de6b562..763f514 100644
--- a/chrome/browser/ui/webui/download_shelf/download_shelf_ui_unittest.cc
+++ b/chrome/browser/ui/webui/download_shelf/download_shelf_ui_unittest.cc
@@ -22,7 +22,8 @@
 
 class TestDownloadShelfHandler : public DownloadShelfHandler {
  public:
-  MOCK_METHOD1(GetDownloads, void(GetDownloadsCallback));
+  MOCK_METHOD1(GetDownloads,
+               void(download_shelf::mojom::PageHandler::GetDownloadsCallback));
   MOCK_METHOD3(ShowContextMenu,
                void(uint32_t download_id, int32_t client_x, int32_t client_y));
   MOCK_METHOD1(DoShowDownload, void(DownloadUIModel*));
